src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs

```diff
@@ -0,0 +1,212 @@
+                        Assert.True(10 == ((System.Data.SqlTypes.SqlInt32)value2).Value, "Unexpected Value");
+                        if (SqlCommandColumnEncryptionSetting.ResultSetOnly == commandSetting)
+                        {
+                            // For ResultSetOnly we don't expect to see plaintext for return values
+                            Assert.True(param.Value is System.Data.SqlTypes.SqlBinary, "Unexpected Return value");
+                        }
+                        else
+                        {
+                            Assert.True(param.Value is System.Data.SqlTypes.SqlInt32, "Unexpected Return value");
+                        }
+                        break;
+                    case SqlCommandColumnEncryptionSetting.UseConnectionSetting:
+                        // Examine the connection string setting to figure out what to expect
+                        if (ConnectionStringSetting.Enabled == connStringSetting)
+                        {
+                            // Expect int
+                            Assert.True(value1 is System.Data.SqlTypes.SqlInt32, "Unexpected type");
+                            Assert.True(value2 is System.Data.SqlTypes.SqlInt32, "Unexpected type");
+                            Assert.True(10 == ((System.Data.SqlTypes.SqlInt32)value2).Value, "Unexpected Value");
+                            Assert.True(param.Value is System.Data.SqlTypes.SqlInt32, "Unexpected Return value");
+                        }
+                        else
+                        {
+                            // Expect varbinary
+                            Assert.True(value1 is System.Data.SqlTypes.SqlBinary, "Unexpected type");
+                            Assert.True(value2 is System.Data.SqlTypes.SqlBinary, "Unexpected type");
+                            Assert.True(param.Value is System.Data.SqlTypes.SqlBinary, "Unexpected Return value");
+                        }
+                        break;
+                }
+            }
+        }
+
+        private void DropFunctions(SqlConnection sqlConnection)
+        {
+            string[] funcNames = {UdfName, UdfNameNotNull};
+
+            foreach (string funcName in funcNames)
+            {
+                using (SqlCommand cmd = new SqlCommand(string.Format("IF EXISTS (SELECT * FROM sys.objects WHERE name = '{0}') \n DROP FUNCTION {0}", funcName), sqlConnection))
+                {
+                    cmd.ExecuteNonQuery();
+                }
+            }
+        }
+
+        public void Dispose()
+        {
+            foreach (string connStrAE in DataTestUtility.AEConnStringsSetup)
+            {
+                using (SqlConnection sqlConnection = new SqlConnection(connStrAE))
+                {
+                    sqlConnection.Open();
+                    Table.DeleteData(fixture.SqlNullValuesTable.Name, sqlConnection);
+                    DropFunctions(sqlConnection);
```

Comment:
Please call `Drop()` on "Table" `fixture.SqlNullValuesTable` in the end.

Reply:
The fixture should already be dropping the table when it is disposed by xunit. This just looks like it is cleaning up the additional data the test is writing to it.

Reply:
This table will be dropped when the tests are finished. Actually, all the tables created by SQLSetupStrategy will be dropped automatically. 
See [public void Dispose()](https://github.com/dotnet/SqlClient/blob/master/src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs)

Reply:
Yes, that's right. Thanks for reminding!